runtime/commonMain/src/kotlinx/serialization/internal/Tagged.kt

```diff
@@ -214,7 +215,13 @@ public abstract class TaggedDecoder<Tag : Any?> : Decoder,
 
     // ---- Implementation of low-level API ----
 
-    final override fun decodeNotNullMark(): Boolean = decodeTaggedNotNullMark(currentTag)
+    // TODO this method should be overridden by any sane format that supports top-level nulls
```

Comment:
I am very happy we've decided to make this class internal and do not provide any stability :)